EapWebInterface/src/test/java/de/hpi/unicorn/application/pages/input/generator/attributeInput/AttributeInputTest.java

```diff
@@ -0,0 +1,131 @@
+        assertEquals("Default input for FloatAttributeInput (Uniform) changed.", floatAttributeInput.getInputOrDefault(), defaultFloatUniformInput);
+        floatAttributeInput.setSelectedMethod(AttributeInput.ProbabilityDistributionEnum.NORMAL);
+        assertEquals("Default input for FloatAttributeInput (Normal) changed.", floatAttributeInput.getInputOrDefault(), defaultFloatNormalInput);
+        userInput = "1.23";
+        floatAttributeInput.setInput(userInput);
+        assertEquals("User input for FloatAttributeInput not properly returned", floatAttributeInput.getInputOrDefault(), userInput);
+
+        //Date attribute
+        assertEquals("Default input for DateAttributeInput changed.", dateAttributeInput.getInputOrDefault(), defaultDateInput);
+        userInput = "2017/01/22T12:00";
+        dateAttributeInput.setInput(userInput);
+        assertEquals("User input for DateAttributeInput not properly returned", dateAttributeInput.getInputOrDefault(), userInput);
+    }
+
+    public void testHasDifferentMethods() {
+        assertFalse("String input shouldn't have different methods.", stringAttributeInput.hasDifferentMethods());
+        assertTrue("Integer input should have different methods.", integerAttributeInput.hasDifferentMethods());
+        assertTrue("Float input should have different methods.", floatAttributeInput.hasDifferentMethods());
+        assertFalse("Date input shouldn't have different methods.", dateAttributeInput.hasDifferentMethods());
+    }
+
+    public void testGetAttributeInputValidator() {
+        assertTrue("Wrong validator for StringAttributeInput.", stringAttributeInput.getAttributeInputValidator() instanceof RegexValidator);
+
+        integerAttributeInput.setSelectedMethod(AttributeInput.ProbabilityDistributionEnum.UNIFORM);
+        assertTrue("Wrong validator for IntegerAttributeInput (Uniform).", integerAttributeInput.getAttributeInputValidator() instanceof
+                IntegerRangeValidator);
+        integerAttributeInput.setSelectedMethod(AttributeInput.ProbabilityDistributionEnum.NORMAL);
+        assertTrue("Wrong validator for IntegerAttributeInput (Normal).", integerAttributeInput.getAttributeInputValidator() instanceof
+                RegexValidator);
+
+        floatAttributeInput.setSelectedMethod(AttributeInput.ProbabilityDistributionEnum.UNIFORM);
+        assertTrue("Wrong validator for FloatAttributeInput (Uniform).", floatAttributeInput.getAttributeInputValidator() instanceof
+                RegexValidator);
+        floatAttributeInput.setSelectedMethod(AttributeInput.ProbabilityDistributionEnum.NORMAL);
+        assertTrue("Wrong validator for FloatAttributeInput (Normal).", floatAttributeInput.getAttributeInputValidator() instanceof
+                RegexValidator);
+
+        assertTrue("Wrong validator for DateAttributeInput.", dateAttributeInput.getAttributeInputValidator() instanceof DateRangeValidator);
+    }
+
```

Comment:
I added tests for `isInRange` in @984ed30.
`calculateRandomValue` is tested in `testGetInputOrDefault` which calls this method.